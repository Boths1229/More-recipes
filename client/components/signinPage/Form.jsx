import React from 'react';
import avatar from '../../public/images/avatar.png';

const Form = props => (
  <div className="pt-5" style={{ textAlign: 'center' }}>
    <form onSubmit={this.handleSubmit} className="m-x-auto app-login-form card-login">
      <h1 className="signin-text"> More-recipe</h1>
      <section className="">
        <img className="img-fluid rounded-circle img-size" src={avatar} alt="" />
      </section>
      <section className="pb-4 pt-3">
        <input className="form-control" placeholder="Username" />
      </section>
      <section className="pb-4">
        <input className="form-control" placeholder="Password" />
      </section>
      <div>
        <button className="btn btn-block btn-orange">
          <a href="recipe.html" className="signin-btn" id="signn-btn">Sign In</a>
        </button>
      </div>
      <div className="pt-2 pb-4 text-align">
        <a href="Forgotpassword" id="text-dec">Forgotten your password?</a>
      </div>
      <div>
        <hr />
      </div>
      <footer>
        <a href="signup" id="text-dec">Create Account</a>
      </footer>
    </form>
  </div>
);

export default Form;
