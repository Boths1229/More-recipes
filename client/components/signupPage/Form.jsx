import React from 'react';
import PropTypes from 'prop-types';

const Form = (props) => {
  const { onChange, onFocus, handleSubmit } = props;
  return (
    <div className="container form-bg">
      <div className="pt-5" style={{ textAlign: 'center' }}>
        <form className="m-x-auto app-login-form card-login">
          <h1 className="signin-text"> More-recipe</h1>
          <section className="signin-text">
            <h3>Create an account</h3>
          </section>
          <div className="error-message">
            { props.value.responseMessage && `*${props.value.responseMessage}*`}
          </div>
          <section className="pb-4 pt-3">
            <input
              name="name"
              className="form-control"
              placeholder="Name"
              onChange={onChange}
              onFocus={onFocus}
              type="text"
              required
            />
          </section>
          <section className="pb-4">
            <input
              name="username"
              className="form-control"
              placeholder="Username"
              onChange={onChange}
              onFocus={onFocus}
              value={props.value.username}
              type="text"
              required
            />
          </section>
          <section className="pb-4">
            <input
              name="email"
              className="form-control"
              placeholder="Email"
              type="email"
              onChange={onChange}
              onFocus={onFocus}
              value={props.value.email}
              required
            />
          </section>
          <section className="pb-4">
            <input
              name="password"
              className="form-control"
              placeholder="Password"
              type="password"
              onChange={onChange}
              onFocus={onFocus}
              value={props.value.password}
              required
            />
          </section>
          <section className="pb-4">
            <input
              name="confirmPassword"
              className="form-control"
              placeholder="Confirm password"
              type="password"
              onChange={onChange}
              onFocus={onFocus}
              value={props.value.confirmPassword}
              required
            />
          </section>
          <div>
            <p>
              <small>
                By clicking &#39;
                Create Account&#39;,
                you are agreeing to our
                <a
                  href="Policy"
                  id="text-dec"
                > Policy
                </a>
                <span>  and  </span>
                <a href="terms" id="text-dec">
                  Terms &amp; Conditions
                </a>
              </small>
            </p>
          </div>
          <div>
            <button
              onClick={handleSubmit}
              className="btn btn-block btn-orange"
            >
              Create Account
            </button>
          </div>
          <div style={{ textAlign: 'left' }}>
            <a href="f" id="text-dec">Forgotten your password?</a>
          </div>
          <div>
            <hr />
          </div>
          <footer>
            <a href="signin">Sign In</a>
          </footer>
        </form>
      </div>
    </div>
  );
};

// Form.propTypes = {
//   value: PropTypes.
// }

export default Form;
