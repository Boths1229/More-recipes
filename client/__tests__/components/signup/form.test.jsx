import React from 'react';
import { shallow } from 'enzyme';
import Form from '../../../components/signupPage/Form';

describe('Form', () => {
  it('matches snapshot', () => {
    const props = {
      value: {}
    };
    const output = shallow(<Form {...props} />);
    expect(output).toMatchSnapshot();
  });
  it('calls onChange props', () => {
    const props = {
      value: {},
      onChange: jest.fn()
    };
    const output = shallow(<Form {...props} />);
    const passwordInputField = output.find('input[name="password"]');
    passwordInputField.simulate('change');
    expect(props.onChange).toHaveBeenCalled();
  });
});
