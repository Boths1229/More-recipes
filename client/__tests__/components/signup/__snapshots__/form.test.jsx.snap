// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="pt-5 text-center"
  >
    <form
      className="m-x-auto app-login-form card-login"
    >
      <h1
        className="signin-text"
      >
         More-recipe
      </h1>
      <section
        className="signin-text"
      >
        <h3>
          Create an account
        </h3>
      </section>
      <div
        className="error-message signup"
      />
      <section
        className="pb-4 pt-3"
      >
        <input
          className="form-control"
          name="name"
          placeholder="Name"
          required={true}
          type="text"
          value=""
        />
      </section>
      <section
        className="pb-4"
      >
        <input
          className="form-control"
          name="username"
          placeholder="Username"
          required={true}
          type="text"
          value=""
        />
      </section>
      <section
        className="pb-4"
      >
        <input
          className="form-control"
          name="email"
          placeholder="Email"
          required={true}
          type="email"
          value=""
        />
      </section>
      <section
        className="pb-4"
      >
        <input
          className="form-control"
          name="password"
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
      </section>
      <section
        className="pb-4"
      >
        <input
          className="form-control"
          name="confirmPassword"
          placeholder="Confirm password"
          required={true}
          type="password"
          value=""
        />
      </section>
      <div>
        <p>
          <small>
            By clicking ' Create Account', you are agreeing to our
            <a
              href="Policy"
              id="text-dec"
            >
               Policy
            </a>
            <span>
                and  
            </span>
            <a
              href="terms"
            >
              Terms & Conditions
            </a>
          </small>
        </p>
      </div>
      <div>
        <button
          className="btn btn-block btn-orange"
        >
          Create Account
        </button>
      </div>
      <div
        className="text-left sign"
      >
        <a
          href="f"
          id="text-dec"
        >
          Forgotten your password?
        </a>
      </div>
      <div>
        <hr />
      </div>
      <footer>
        <Link
          replace={false}
          to="/signin"
        >
          Sign In
        </Link>
      </footer>
    </form>
  </div>
</div>
`;
